documentation: fix typos

FardaleM opened the pull request:
Fix typo in the docstring of `starts_with` and `ends_with`.

nojb reviewed (dismissed):
LGTM

dra27 commented:
It's failing CI, @nojb! (I'm just "reviewing")

dra27 commented on stdlib/stringLabels.mli:
```diff
@@ -151,7 +151,7 @@ val compare : t -> t -> int
 val starts_with :
   prefix (* comment thwarts tools/sync_stdlib_docs *) :string -> string -> bool
-(** [starts_with ][~][prefix s] is [true] if and only if [s] starts with
+(** [starts_with ~prefix s] is [true] if and only if [s] starts with
```
This line was correct - the same pattern is needed with `ends_with` below (and then _the same_ text appears in `string.mli`)

FardaleM commented:
I am also looking at the CI. The fail is from `tools/sync_stdlib_docs` which tries to remove a `~`in front of a label argument for `starts_with` and `ends_with`. To me it looks like a mistake from the script but I don't see a trivial way of fixing it.

dra27 reviewed (changes requested):
You should be able to run `tools/sync_stdlib_docs` after editing `string/stringLabels.mli` to test the changes (slightly faster than playing CI "golf"). There are very few functions which are labelled in both the labelled and unlabelled versions of the modules. At present, the script is made slightly simpler by using that convoluted syntax (although it does also show up in the rendering, so it's not at all perfect).

Improvements to the script are very welcome 🙂

FardaleM committed:
documentation: improve the sync_stdlib_docs tool

Try to improve the case when there is function with label argument in a
non label module.

FardaleM commented:
I tried something slightly different to improve a bit the situation. Tell me if that's OK or if I should revert to the previous solution.

dra27 reviewed (approved):
This is an unquestionable improvement, thank you! The .mli files are less cluttered with this trick, and now in the rendered version the `~` is correctly right next to the label name. There remains still a tiny bit of extra space between the two `<code>` blocks (at least with ocaml.org's stylesheet), but this is great as it stands
